compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -564,6 +564,37 @@ fn transcribe_metavar_expr<'a>(
         span
     };
     match *expr {
+        MetaVarExpr::Concat(ref elements) => {
+            let mut concatenated = String::new();
+            for element in elements.into_iter() {
+                let s = 'string: {
+                    if !element.is_var {
+                        break 'string element.ident.to_string();
+                    }
+                    let span = element.ident.span;
+                    let mrni = MacroRulesNormalizedIdent::new(element.ident);
+                    if let Some(nm) = lookup_cur_matched(mrni, interp, &repeats)
+                        && let MatchedNonterminal(nt) = nm
+                    {
+                        if let Nonterminal::NtIdent(nt_ident, _) = &nt.0 {
+                            nt_ident.to_string()
+                        } else {
+                            return Err(cx.dcx().struct_span_err(
+                                span,
+                                "`${concat(..)}` currently only accepts identifiers as parameters",
+                            ));
+                        }
+                    } else {
+                        element.ident.to_string()
+                    }
```

Comment:
is this right? `element.is_var` implies that that was `$ident` so we should never produce `element.ident.to_string()`...

Reply:
You are right. This branch has been removed.